In [8]:
import pandas as pd
arr = pd.read_csv("arrestdata.tsv", sep="\t")

In [10]:
count = {}
for index, row in arr.iterrows():
    name = row["INCIDENTNEIGHBORHOOD"]
    qty = 1
    if name in count:
        count[name] = count[name]+qty
    else:
        count[name] = qty

print(count)


{'Bloomfield': 984, 'Outside City': 316, 'Westwood': 226, 'East Allegheny': 2217, 'Crafton Heights': 601, 'Greenfield': 537, 'Brookline': 1390, 'Regent Square': 37, 'East Liberty': 1889, nan: 2489, 'East Hills': 1194, 'Carrick': 2352, 'Allegheny Center': 1244, 'South Side Flats': 3497, 'Sheraden': 1285, 'East Carnegie': 48, 'Golden Triangle/Civic Arena': 83, 'Point Breeze North': 289, 'Perry South': 1057, 'Allentown': 947, 'North Shore': 814, 'Marshall-Shadeland': 1516, 'Central Oakland': 521, 'Middle Hill': 932, 'Strip District': 503, 'Lincoln-Lemington-Belmar': 1362, 'Central Northside': 23, 'Troy Hill-Herrs Island': 6, 'Highland Park': 558, 'Homewood North': 1969, 'Northview Heights': 713, 'Brighton Heights': 1114, 'Hazelwood': 1396, 'Perry North': 587, 'Bedford Dwellings': 665, 'Elliott': 663, 'Beechview': 1201, 'Fineview': 796, 'Glen Hazel': 169, 'Spring Hill-City View': 711, 'Bluff': 1502, 'Manchester': 471, 'Homewood West': 764, 'Outside County': 57, 'Point Breeze': 301, 'Homewo

In [33]:
arr['INCIDENTNEIGHBORHOOD'] = arr['INCIDENTNEIGHBORHOOD'].replace({'Mt. Oliver Boro' : 'Mt. Oliver'})
arr['INCIDENTNEIGHBORHOOD'] = arr['INCIDENTNEIGHBORHOOD'].replace({'Mt. Oliver Neighborhood' : 'Mt. Oliver'})
arr['INCIDENTNEIGHBORHOOD'] = arr['INCIDENTNEIGHBORHOOD'].replace({'Mount Oliver' : 'Mt. Oliver'})
neighborhood_counts = arr['INCIDENTNEIGHBORHOOD'].value_counts().sort_values()
neighborhood_counts_df = neighborhood_counts.reset_index()
neighborhood_counts_df.columns = ['INCIDENTNEIGHBORHOOD', 'Count']
pd.set_option('display.max_rows',None)
print(neighborhood_counts_df)

           INCIDENTNEIGHBORHOOD  Count
0        Troy Hill-Herrs Island      6
1                   Mt. Oliver      18
2             Central Northside     23
3                     Ridgemont     37
4                 Regent Square     37
5                 New Homestead     39
6                Swisshelm Park     43
7                Chartiers City     46
8                 East Carnegie     48
9                     St. Clair     55
10               Outside County     57
11                Outside State     67
12                  Summer Hill     76
13                      Oakwood     82
14  Golden Triangle/Civic Arena     83
15                         Hays    120
16                      Windgap    124
17                   Mt. Oliver    126
18                    Fairywood    127
19               Allegheny West    130
20            Arlington Heights    155
21                  Polish Hill    166
22                   Glen Hazel    169
23                   Friendship    174
24                  Morni

From the data we can see that the top 5 places with the least arrests are:
 - Central Northside (23)
 - Regent Square (37)
 - Ridgemont (37)
 - Homestead (39)
 - Chartiers City (46)
 - East Carnegie (48)
 - St. CLair (55)
 - Golden Triangle/Civic Arena (83)
 - Allegheny West (130)
 - Mount Oliver (144)


In my code, first I created a list to keep track of all neighborhoods and the amount of arrests there were. Then, I made a loop that goes through each individual piece of the data and keeps track of that neighborhood. If the neighborhood is not in the list yet, it adds the name of the neighborhood to the list and makes the count for arrests there 1. If it already is in the list then the count for that neighborhood goes up by 1. It does this throughout the entire data table, thus giving us the final results. I then printed the list so that we could see these results. Based on the data, we can see that the 'safest' neighborhood is Central Northside, followed by a tie between Regent Square and Ridgemont.